test/integration/model/findByPks.test.js

```diff
@@ -0,0 +1,82 @@
+'use strict';
+
+const chai = require('chai'),
+  Sequelize = require('../../../index'),
+  expect = chai.expect,
+  Support = require('../support'),
+  DataTypes = require('../../../lib/data-types'),
+  current = Support.sequelize;
+
+describe(Support.getTestDialectTeaser('Model'), () => {
+  beforeEach(function() {
+    this.User = this.sequelize.define('User', {
+      username: DataTypes.STRING,
+      secretValue: DataTypes.STRING,
+      data: DataTypes.STRING,
+      intVal: DataTypes.INTEGER,
+      theDate: DataTypes.DATE,
+      aBool: DataTypes.BOOLEAN,
+      binary: DataTypes.STRING(16, true)
+    });
+
+    return this.User.sync({ force: true });
+  });
+
+  describe('findByPks', () => {
+    if (current.dialect.supports.transactions) {
+      it('supports transactions', function() {
+        return Support.prepareTransactionTest(this.sequelize).then(sequelize => {
+          const User = sequelize.define('User', { 
+            id: {
+              type: Sequelize.INTEGER,
+              primaryKey: true
+            },
+            username: Sequelize.STRING 
+          });
+
+          return User.sync({ force: true }).then(() => {
+            return sequelize.transaction().then(t => {
+              return User.create({ id: 1, username: 'foo' }).then(() => {
+                return User.create({ id: 2, username: 'bar' }).then(() => {
+                  return User.findByPks([1, 2]).then(users => {
+                    expect(users.length).to.equal(2);
+                    return t.rollback();
+                  });
+                });
+              });
+            });
+          });
+        });
+      });
+    }
+
+    it('should not crash on an empty where array', function() {
+      return this.User.findByPks([]);
+    });
```

Comment:
This test is OK, but I think you can improve it by checking that the result is an empty array (instead of just checking if it doesn't throw)